dropgz/pkg/embed/payload.go

```diff
@@ -81,14 +84,29 @@ func Extract(path string) (*compoundReadCloser, error) {
 	return &compoundReadCloser{closer: f, readcloser: r}, nil
 }
 
-func deploy(src, dest string) error {
+func deploy(log *zap.Logger, src, dest string) error {
 	rc, err := Extract(src)
 	if err != nil {
 		return err
 	}
 	defer rc.Close()
 	// check if the file exists at dest already and rename it as an old one
 	if _, err := os.Stat(dest); err == nil {
+		// For windows we need to close the process running with the binary before we can rename it.
+		// This is because the file is locked by the process.
+		// We can't use the os.Rename() function because it will fail with an error.
+		if runtime.GOOS == "windows" {
```

Comment:
@rbtr @tamilmani1989 Need review of this part of code. Added the comment on why we need it.

Reply:
ya this is bad..i guess its `azure-vnet-telemetry` process.. i'm considering it to move to cns to start telemetry service.. then we dont need to copy `azure-vnet-telemetry` for swift

Reply:
can we remove this code ? Lets consider telemetry process not running when we use dropgz.. for cniv1, lets update cniv1 installer deployment yaml to kill azure-vnet-telemetry process before invoking dropgz

Reply:
we should also give a retry if copy fails due to transient error

Reply:
> we should also give a retry if copy fails due to transient error 

disagree, the init container will be restarted if it exits with an error

Reply:
in that case, its fine